streamlit_demo_apps/README.md

```diff
@@ -10,40 +10,49 @@ Deployed Streamlit apps live here in the streamlit_demo_apps folder.
 cp .streamlit/secrets.toml.example .streamlit/secrets.toml
 ```
 
-### 1. Create an app file in the streamlit_demo_apps folder
+### 1. Run the script to generate a new streamlit app for your program
 
-Check out the addition app file example:
+From the root folder of this repo, run the generate-streamlit-app script:
 
-`app_addition.py`
+```
+python3 generate-streamlit-app.py
+```
+
+### 2. Follow the prompts to
 
-### 2. Copy the compiled Nada program files from the target/ folder into the streamlit_demo_apps/compiled_nada_programs folder
+- Select an existing program (from the src/ directory)
+- Select an existing yaml test file for your program (from the tests/ directory)
 
-Check out the compiled Nada program files for addition:
+This will generate a Streamlit app file: streamlit*demo_apps/app*[your_program_name].py. The script will run the Streamlit app locally with this command
 
-nada binary `addition.nada.bin`
-nada json `addition.nada.json`
+```
+streamlit run streamlit_demo_apps/app_[your_program_name].py`
+```
 
-### 3. Update your app file with the corresponding program name and program test name
+### 3. Test your Streamlit app locally
 
-Check out the addition app file example:
+View the app in your browser to make sure everything works as expected.
 
-`app_addition.py`
+### 4. Deploy your app
+
+Add and commit your new streamlit app code to your forked Github repo. (Code must be connected to a remote, open source GitHub repository to deploy a Streamlit app.)
 
 ```
-program_name = 'addition'
-program_test_name = 'addition_test'
+git add .
+git commit -m "my new streamlit nillion app"
+git push origin main
 ```
 
-### 4. Test your Streamlit app locally
+Once you've committed the open source code, you can click the "deploy" button within your local streamlit app. Sign in with Github and select the "Deploy Now" on Streamlit Community Cloud option to deploy the app for free.
 
-Make sure the apps will work when deployed by testing this command from the root folder.
+#### Deploy an app settings
```

Comment:
What do you mean "an app settings"? It reads a bit odd.

Reply:
good point I meant "settings for deploying an app" but I can make this clearer in the title